### DISCUSSION TITLE
Experiences with Setting Up gaiaCore

### DISCUSSION THREAD

**Discussion by TheCedarPrince** [OHDSI/GIS#322] [General]
Hey folks!

Just wanted to mention some of my experiences with setting up GAIA locally on my computer (Fedora 38):

1. Lots of additional dependencies were needed when building gaiaCore that I was surprised about; here was a list I made of what I needed while installing fedora dependencies: 

    1. `gdal-devel`
    2. `proj-devel`
    3. `geos-devel`
    4. `postgresql-devel`
    5. `docker`

Then this should in theory work for installing all of the R packages needed.

2. Wasn't too much of a hang up but it wasn't explicit that you needed to create a `gaiadb` database on your [postgres instance](https://ohdsi.github.io/GIS/ht-get-started.html).
3. The handy R package `openadds` has been deprecated since the end of 2023 making it rather challenging to follow the tutorial. Not sure what would be a good alternative here. I had to use an old version of the package not on CRAN to follow along.
4. I was surprised about the Docker dependency and that it was needed as it wasn't mentioned as a dependency. I couldn't get the code here working until I ran R with superuser permissions which feels a bit uncomfortable. 

Those were some of the things I experienced in the install -- thought I'd share in case it would be useful to folks. Thanks for great work on this all! :smile:

**Comment by kzollove** [comment]
This is really helpful, thanks for giving us this feedback @TheCedarPrince .

> Lots of additional dependencies were needed when building gaiaCore

Thanks for pointing this out, we should make this explicit in our docs. I might list docker as an optional dependency, unless you found it was necessary outside of geocoding.

> it wasn't explicit that you needed to create a gaiadb database on your postgres instance.

Hm, we'll need to make it explicit

> The handy R package openadds has been deprecated since the end of 2023

Did this factor in during the geocoding workflow? 

> Docker dependency

Yes, its not ideal. Essentially we want to offer our "wrapped" version of DeGauss as a potential geocoding solution. We could potentially rebuild DeGauss and just not have it wrapped in Docker, but then it becomes another OHDSI GIS project to maintain... 



Thanks again for this super valuable feedback. I will keep you updated as we start addressing these.

**Reply by TheCedarPrince** [reply]
> I might list docker as an optional dependency, unless you found it was necessary outside of geocoding.

Yea I should've specified that docker wasn't needed outside of geocoding. Optional sounds reasonable.

> Did this factor in during the geocoding workflow?

For general geocoding, no. But for users following the beginner tutorial, this might entirely derail them as I had to go down a rabbit hole for about 20 minutes to find a non-CRAN version of this package (rdrr.io version) and then discovered that one of its dependencies was also archived (maptools) leading to another rabbit hole. So, I saw this as rather major-ish.

> We could potentially rebuild DeGauss and just not have it wrapped in Docker, but then it becomes another OHDSI GIS project to maintain...

Yea, that's very fair. I was exploring DeGauss and saw that it seems like a strong hodge podge of a variety of tools (Ruby, R, C, shell even). I was seeing if I could build it locally on my machine and although I could, it just would make the entire process a bit more challenging. 

> Thanks again for this super valuable feedback. I will keep you updated as we start addressing these.

Sounds great! Let me know if you have any more questions or thoughts!
